web3/providers/ipc.py

```diff
@@ -18,10 +26,12 @@
 try:
     from json import JSONDecodeError
 except ImportError:
-    JSONDecodeError = ValueError
+    # type ignored b/c mypy unhappy w/ conflicting exception types
+    # difficult to cast / resolve w/ conditional import
+    JSONDecodeError = ValueError  # type: ignore
```

Comment:
Worth checking if this is still necessary.  Docs say it was introduced in 3.5 which suggests we may not need this anymore.